app/doi.py

```diff
@@ -47,11 +57,15 @@ def mint_doi(event, _context, _kwargs):
     else:
         # DataCite successful response *would* have all our repo info,
         # so extract just the newly minted DOI for the response body
-        return {
-            "statusCode": 201,
-            "body": json.dumps(
+        if method == "POST":
+            return return_response.update(
```

Comment:
pitfall I personally stumble into constantly: python's dict.update is inplace, so as-is this would return None
```suggestion
            return_response.update(...)
        ...
        return return_response
```